imap_processing/ialirt/l0/process_swe.py

```diff
@@ -0,0 +1,246 @@
+import pandas as pd
+import xarray as xr
+from numpy.typing import NDArray
+
+from imap_processing.ialirt.utils.grouping import find_groups
+from imap_processing.swe.l1a.swe_science import decompressed_counts
+from imap_processing.swe.l1b.swe_l1b_science import (
+    deadtime_correction,
+    read_in_flight_cal_data,
+)
+from imap_processing.swe.utils.swe_constants import (
+    ESA_VOLTAGE_ROW_INDEX_DICT,
+    GEOMETRIC_FACTORS,
+)
+from imap_processing.swe.utils.swe_utils import combine_acquisition_time
+
+logger = logging.getLogger(__name__)
+
+
+def decompress_counts(raw_counts: NDArray) -> NDArray:
+    """
+    Perform decompression of raw counts using a predefined decompression table.
+
+    Parameters
+    ----------
+    raw_counts : np.ndarray
+        Array of raw compressed counts with shape (n_time, n_cem, n_energy_step).
+
+    Returns
+    -------
+    counts : np.ndarray
+        Array of decompressed counts with the same shape as raw_counts.
+    """
+    decompression_table = np.array([decompressed_counts(i) for i in range(256)])
+
+    # Decompress using the precomputed table
+    counts = decompression_table[raw_counts]
+
+    return counts
+
+
+def phi_to_bin(phi_values: NDArray) -> NDArray:
+    """
+    Convert phi values to corresponding bin indices.
+
+    Parameters
+    ----------
+    phi_values : NDArray
+        Array of phi values.
+
+    Returns
+    -------
+    bin_indices : NDArray
+        Array of bin indices.
+    """
+    # Ensure it wraps correctly within 0-29 bins
+    return ((phi_values - 12) // 12) % 30
+
+
+def prepare_raw_counts(grouped: xr.Dataset, cem_number: int = 7) -> NDArray:
+    """
+    Reformat raw counts into a 3D array binned by phi.
+
+    Parameters
+    ----------
+    grouped : xr.Dataset
+        Dataset containing grouped i-ALiRT packet data for 1 minute.
+    cem_number : int
+        Number of CEMs (default 7).
+
+    Returns
+    -------
+    raw_counts : NDArray
+        Raw counts with shape (8, 7, 30).
+
+    Notes
+    -----
+    Array of raw counts with shape (8, 7, 30), where:
+        - 8 corresponds to the 8 energy steps.
+        - 7 corresponds to the 7 CEM detectors.
+        - 30 corresponds to the 30 phi bins.
+    """
+    raw_counts = np.zeros((8, cem_number, 30), dtype=np.uint8)
+
+    # Compute phi values and their corresponding bins
+    phi_values = np.array(
+        [
+            (12 + 24 * grouped["swe_seq"].values) % 360,  # Energy steps 0 and 1
+            (24 + 24 * grouped["swe_seq"].values) % 360,  # Energy steps 2 and 3
+        ]
+    )
+    phi_bins = phi_to_bin(phi_values).astype(int)  # Get phi bin indices
+
+    # Energy bin lookup table (indexed by quarter cycle)
+    energy_bins = np.array(
+        [
+            [1, 5, 3, 7],  # 0-14 (first quarter cycle)
+            [2, 6, 0, 4],  # 15-29 (second quarter cycle)
+            [3, 7, 1, 5],  # 30-44 (third quarter cycle)
+            [0, 4, 2, 6],  # 45-59 (fourth quarter cycle)
```

Comment:
I believe it's in this order.
```suggestion
            [1, 5, 7, 3],  # 0-14 (first quarter cycle)
            [2, 6, 4, 0],  # 15-29 (second quarter cycle)
            [3, 7, 5, 1],  # 30-44 (third quarter cycle)
            [0, 4, 6, 2],  # 45-59 (fourth quarter cycle)
```

Reply:
Fixed.